src/auth.ts

```diff
@@ -0,0 +1,45 @@
+/*
+auth.ts - Provides auth related methods for the Bluebutton class
+*/
+
+import BlueButton from "./index";
+import {
+  generateCodeChallenge,
+  generateRandomState,
+} from "./utils/generatePKCE";
+
+export function setCallBackUrl(this: BlueButton, callBackUrl: string): any {
```

Comment:
```suggestion
export function setCallBackUrl(this: BlueButton, callBackUrl: string) {
```
For functions that don't return anything, we'll actually want to use `void`. Typescript does a pretty good job of inferring the typing though, so if you don't have a `return` statement, it knows that it is just void. If you are using VSCode, you can hover over function/variable names and it'll show you the inferred typings

Reply:
@nbragdon Yikes. Knew better... Forgot to remove that. Needed it for the other function, then did a copy to start a new one an didn't remove :man_facepalming: afterward.

Reply:
@nbragdon Related to the `this` usage and having the function code in another file, I like your example better. That is less confusing.  Updating now...